app/src/main/java/org/oppia/app/profile/AddProfileActivity.kt

```diff
@@ -30,7 +30,9 @@ class AddProfileActivity : InjectableAppCompatActivity() {
   }
 
   override fun onSupportNavigateUp(): Boolean {
-    finish()
```

Comment:
Why isn't finish sufficient?

Reply:
The admin pin won't get updated when the user goes through the admin pin flow unless we start the activity again because of the live data bug.

Reply:
Ack. Similar to below--please document that here to make it explicit.

Reply:
Updated with comment.